test/integration/universe/wizards-v1/universe-lifecycle/basic-paths/scenario-2.ts

```diff
@@ -0,0 +1,257 @@
+      },
+    );
+    await Erc721ConfigurablePresetValidator.validateWarperState(
+      deployedWarperAddress,
+      originalCollection,
+      {
+        availabilityPeriodStart: 0,
+        availabilityPeriodEnd: MAX_TIME_PROTOCOL_UINT32,
+      },
+      { minRentalPeriod: 0, maxRentalPeriod: MAX_TIME_PROTOCOL_UINT32 },
+      { metahub, reader: stranger },
+    );
+    await WarperManagerValidator.validateWarperRegistration(
+      setupUniverseSuccessTx,
+      universeId,
+      deployedWarperAddress,
+      originalCollection.address,
+      ASSET_CLASS_IDS.ERC721,
+      { warperManager },
+    );
+    await TaxTermsRegistryValidator.validateUniverseWarperTaxTermsRegistration(
+      setupUniverseSuccessTx,
+      universeId,
+      deployedWarperAddress,
+      taxTerms,
+      { taxTermsRegistry },
+    );
+  });
+
+  it(`${TEST_CASE(TEST_OK)} Setup Universe with a pre-deployed Warper`, async () => {
+    const { deployedWarperAddress } = await new WarperDeployFromPresetBuilder(warperPresetFactory, universeOwner)
+      .withWarperConstructor(warperConstructor)
+      .deployFromPresetWithSuccess();
+    const universeBuilder = new UniverseWithErc721WarperBuilderV1(universeWizardV1, universeOwner)
+      .withUniverseConstructor(universeConstructor)
+      .withWarperConstructor(warperConstructor)
+      .withWarperAddress(deployedWarperAddress)
+      .withTaxTerms(taxTerms);
+    const { setupUniverseSuccessTx, universeId } = await universeBuilder.setupUniverseWithSuccessAndTx();
+
+    await UniverseValidatorV1.validateUniverseSetupProperly(
+      setupUniverseSuccessTx,
+      universeId,
+      universeOwner,
+      universeConstructor.getUniverseParamsSafely(),
+      {
+        universeRegistry,
+        reader: stranger,
+      },
+    );
+    await WarperManagerValidator.validateWarperRegistration(
+      setupUniverseSuccessTx,
+      universeId,
+      deployedWarperAddress,
+      originalCollection.address,
+      ASSET_CLASS_IDS.ERC721,
+      { warperManager },
+    );
+    await TaxTermsRegistryValidator.validateUniverseWarperTaxTermsRegistration(
+      setupUniverseSuccessTx,
+      universeId,
+      deployedWarperAddress,
+      taxTerms,
+      { taxTermsRegistry },
+    );
+  });
+
+  it(`${TEST_CASE(TEST_NOT_OK)} Setup Universe with an empty name`, async () => {
+    universeConstructor = universeConstructor.withUniverseName('');
+    const setupUniverseTx = new UniverseWithErc721WarperBuilderV1(universeWizardV1, universeOwner)
+      .withUniverseConstructor(universeConstructor)
+      .withWarperConstructor(warperConstructor)
+      .withTaxTerms(taxTerms)
+      .setupUniverseWithTxPromiseOnly();
+
+    await expect(setupUniverseTx).to.be.revertedWithCustomError(
+      { interface: IUniverseRegistry__factory.createInterface() },
+      UNIVERSE_REGISTRY_CUSTOM_ERRORS.EMPTY_UNIVERSE_NAME,
+    );
+  });
+
+  it(`${TEST_CASE(TEST_NOT_OK)} Setup Universe with no payment tokens`, async () => {
```

Comment:
An idea, to move those bad cases to scenario-3 and differentiate scenario-2 and scenario-3 by first being successful/happy paths and second one failure paths

Reply:
Then we probably should do the same for scenario-1?

Reply:
I think, we have a flexibility. I am just giving my vote for compact scnarios with minimum viable interfaces. Sometimes it might be an overhead to split to 2 separate scenarios, when you have just 2 test cases. In your piece - you have more and we could improve the readability!